### PR TITLE
Feature/movie

### PR DESCRIPTION
**Pull request by aspitz1**
#10 
This PR makes a Movie component that shows the details of a movie.

I decided it may be best to store the movie object in the App, then pass it through a prop to the Movie component. This enables the movie component to be a functional component. I think this is good because the Movie component doesn't have interactivity at this time. It also allows the App component to be the only component doing API calls. I was thinking it may be good to have those in one place. For now, the Movie data is hard coded into the state of the App.
Let me know if you are thinking there may be a more ideal way to do this!

I also included a boolean value in the state of the App that is set to false. This can be switched to true to display the Movie component. When this PR is initially pulled down and run the Movie component will not show. This will be the case til the boolean is true.

This PR also includes minimal CSS to style the Movie component. It pretty much just handles layout and image sizing. The only fancy thing is the vignette on the movie backdrop image.  This background image is being handled inline because the URL comes in the API call. There isn't a way that I know of to dynamically pass that to the CSS other than inline. Here is a [blog post](https://www.pluralsight.com/guides/inline-styling-with-react) on inline styles with React. I had to do some research so I thought I would include a resource for it.

I imagine this PR will create conflicts with the AllMovies component in the App, so it is possible we may want to wait to merge this into the main til the AllMovies component is merged. I don't know if it really matters b/c there will probably be conflicts no matter what.

### CODE REVIEW

**Comment by aspitz1** [comment]
I accidentally pushed the Navbar into this PR. I removed the commits immediately. It shouldn't be an issue.  I almost left it, but it seemed like something that was a safe thing to do at this point.

**Comment by aspitz1** [comment]
I am seeing now my PR wasn't as thorough as yours in regards to lines of code! I will follow your lead on my next PR but please leave any feedback on how I could improve my PR writing too!

File: src/App/App.js [hunk 1]
```diff
@@ -1,19 +1,40 @@
...
+          'Crime',
+          'Drama',
+          'Thriller'
+        ],
+        budget: 0,
+        revenue: 152812,
+        runtime: 96,
+        tagline: 'Kill. Or be killed.',
+        averageRating: 5.875
+      }
```
Comment (mayakappen) [inline src/App/App.js:30]: Cool! Is app going to hold just one movie at a time?

File: src/App/App.js [hunk 2]
```diff
@@ -1,19 +1,40 @@
...
-    <div className="App">
-      <h1>Hello World!</h1>
-    </div>
-  );
-}
+
+  render() {
+    return (
+      <div className="App">
+        {this.state.isMovieView && <Movie movieDetails={this.state.movie}/>}
```
Comment (mayakappen) [inline src/App/App.js:38]: I will try and add my component call above this? I'll let you know if I run into problems

File: src/Movie/Movie.css [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    margin: 0;
+    padding-left: 50px;
+    font-size: 3em;
+}
+
+.movieHeadingDetails {
+    list-style: none;
+}
+
+.movie-description-container {
```
Comment (mayakappen) [inline src/Movie/Movie.css:19]: so many containers is that your secret?

File: src/Movie/Movie.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import './Movie.css'
+
+const Movie = ({ movieDetails }) => {
+    const {
+        title, posterPath, backdropPath, releaseDate, overview,
+        genres, budget, revenue, runtime, tagline, averageRating
+    } = movieDetails;
```
Comment (mayakappen) [inline src/Movie/Movie.js:7]: ok idk what is going on here but it looks cool! Looks like you're breaking up the movieDetails into those constants? Do the curlies make it know that those are key values? Would love this explained like I'm 5 lol

File: src/Movie/Movie.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+    const releaseDateFormatted = new Date(releaseDate).toLocaleDateString('en-US');
```
Comment (mayakappen) [inline src/Movie/Movie.js:9]: love the toLocalDateString() usage

File: src/Movie/Movie.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    
+    return (
+        <div className='movie-container'>
+            <div className='movie-heading-container' 
+                style={{backgroundImage: `url(${backdropPath})`, 
+                backgroundSize: 'cover',
+                boxShadow: 'inset 30vw 10px 100px black'
```
Comment (mayakappen) [inline src/Movie/Movie.js:16]: are these the inline styles you were telling me about? I love it should I do that on my components too for consistency?

**Review by mayakappen — APPROVED** [review]
Love the look of the page and the logic of everything, great job!

**Pull request merged**